### PR TITLE
Fix return tag scene

### PR DESCRIPTION
**Pull request by callemand**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] Did you test this pull request in real life? With real devices? If this development is a big feature or a new service, we recommend that you provide a Docker image to the community ([french forum](https://community.gladysassistant.com/)/[english forum](https://en-community.gladysassistant.com/)) for testing before merging.

### Description of change

 - Order tag list by alphabetical order
 - Add badge with scene count
 - Add button to unselect all tags
 - Modify and to or into get scene with tag filter
 - Fix "martelage des tags"
Screenshot 2023-11-22 at 11 28 20

### CODE REVIEW

File: server/lib/scene/scene.get.js [hunk 1]
```diff
@@ -58,14 +59,25 @@ async function get(options) {
...
   if (optionsWithDefault.searchTags) {
-    const tags = await db.TagScene.findAll({
-      fields: 'scene_id',
-      where: {
-        [Op.or]: optionsWithDefault.searchTags.split(',').map((tag) => ({ name: { [Op.like]: `%${tag}%` } })),
-      },
-    });
+    const tags = optionsWithDefault.searchTags.split(',');
+
+    const tagsWithSceneId = await Promise.all(
```
Comment (Pierre-Gilles) [inline server/lib/scene/scene.get.js]: Can you do this in one query like before?

File: server/lib/scene/scene.getTag.js [hunk 2]
```diff
@@ -8,8 +8,15 @@ const db = require('../../models');
  * const scenes = await gladys.scene.getTag();
  */
 async function getTag() {
-  const tagScenes = await db.TagScene.findAll({
+  /* const tagScenes = await db.TagScene.findAll({
```
Comment (Pierre-Gilles) [inline server/lib/scene/scene.getTag.js]: comment

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for the quick fixes! I have a few tech feedbacks.. :)

File: server/lib/scene/scene.get.js (hunk 1 above)
Comment (callemand) [reply server/lib/scene/scene.get.js]: If it's or needed no, if it's and needed, I come back to the previous code

Comment (callemand) [reply server/lib/scene/scene.get.js]: Finally, I found a solution with only 1 query but I use more the cpu for the calculation :)

File: server/lib/scene/scene.get.js [hunk 3]
```diff
@@ -58,14 +59,29 @@ async function get(options) {
...
+        },
+      })
+    ).map((tag) => tag.get({ plain: true }));
+
+    const tagsWithSceneId = {};
+    sceneIdsAndNames.forEach((sceneIdAndName) => {
+      if (sceneIdAndName.name in tagsWithSceneId === false) {
+        tagsWithSceneId[sceneIdAndName.name] = [];
+      }
+      tagsWithSceneId[sceneIdAndName.name].push(sceneIdAndName.scene_id);
```
Comment (Pierre-Gilles) [inline server/lib/scene/scene.get.js:78]: No need to do all that, you need to do a JOIN with a simple WHERE :) 

In Sequelize: https://sequelize.org/docs/v6/advanced-association-concepts/eager-loading/#eager-loading-filtered-at-the-associated-model-level

Here you'll need to do n JOIN (one per tag)

File: front/src/routes/scene/style.css [hunk 4]
```diff
@@ -56,3 +56,24 @@
...
+  right: -10px;
+  top: 4px;
+
+  /*
+  background-clip: padding-box;
+  border-radius: 15px;
+  background-color: #F0F0F0;
+  padding: 0px 7px;
+  display: inline;
+
```
Comment (Pierre-Gilles) [inline front/src/routes/scene/style.css]: Comment?

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Hey! Thanks for the changes :) 

I think the query part can be done way more simply, see my comments :) Don't hesitate if you need help on this!

**Review by Pierre-Gilles — DISMISSED** [review]
Good for me! Thanks for the changes! 

Edit: Seems there is a slight bug on mobile ?

Screenshot 2023-11-27 at 15 53 18

**Review by Pierre-Gilles — APPROVED** [review]
Better like this on mobile 👍

**Pull request merged**